### DISCUSSION TITLE
No matter what "precision" or "epsilon" I set, the result I get from "sqrt" is not rounded up

### DISCUSSION THREAD

**Discussion by PaulRBerg** [josdejong/mathjs#2612] [Q&A] [answered]
This is potentially related to https://github.com/josdejong/mathjs/issues/2608.

I have the following configuration:

```typescript
{
    number: "BigNumber",
    precision: 60 + 18 + 2,
    epsilon: 1e-80,
}
```

I am then trying to run `sqrt(x.mul(y))` over the following two numbers:

```typescript
const x = math.bignumber!("115792089237316195423570985008687907853269984665640564039457584007913129639935");
const y = math.bignumber!("0.000000000000000001");
const result = math.sqrt!(x.mul(y));
```

This produces the result `340282366920938463463.374607431768211455`.

But, as per the result I get on Wolfram, I was expecting the result to be `340282366920938463463.374607431768211456` (with a `6` at the end) instead. More specifically, I was expecting the `.99999...` decimals to cause the result to be rounded up.

Why is the result not rounded up, and how can I make it be so?

Note: this started to happen after I upgraded `mathjs` to v10.6.4. When I was using v10.4.0, the value I was getting was `340282366920938463463.374607431768211456`.

**Comment by josdejong** [comment, marked as answer]
Executing your expression, I get:

```
340282366920938463463.37460743176821145599999999999999999999999999999999999999853
```
Which is indeed the correct answer but contains a rounding error.

The `BigNumber` data type (powered by [Decimal.js](https://mikemcl.github.io/decimal.js/)) has a higher precision but is still a floating point number type, and hence can have round-off errors. 

You can use `math.format` or `math.round` to round the result if you want to get rid of the round-off errors:

```js
math.config({precision: 60+18+2})
const x = math.bignumber("115792089237316195423570985008687907853269984665640564039457.584007913129639935")
const y = math.bignumber("0.000000000000000001")
const result = math.sqrt(x.mul(y))
// "340282366920938463463.37460743176821145599999999999999999999999999999999999999853"

const decimals = 18
const roundedResult = math.round(result, decimals)
// "340282366920938463463.374607431768211456" 
```

**Reply by PaulRBerg** [reply]
Thanks for the tip to use `math.round`!
